parsers/cxx/cxx_parser_variable.c

```diff
@@ -568,14 +595,24 @@ bool cxxParserExtractVariableDeclarations(CXXTokenChain * pChain,unsigned int uF
 						return bGotVariable;
 					}
 				} else if(
-						// Still 'MACRO(whatever) variable;' case
+						// Possibly one of:
+						//   MACRO(whatever) variable;
+						//   decltype(whatever) variable;
 						cxxTokenTypeIs(pTokenBefore,CXXTokenTypeParenthesisChain) &&
 						pTokenBefore->pPrev &&
-						cxxTokenTypeIs(pTokenBefore->pPrev,CXXTokenTypeIdentifier) &&
-						!pTokenBefore->pPrev->pPrev
+						!pTokenBefore->pPrev->pPrev &&
+						(
+							// macro
+							cxxTokenTypeIs(pTokenBefore->pPrev,CXXTokenTypeIdentifier) ||
+							// decltype
+							(
+								cxxTokenTypeIs(pTokenBefore->pPrev,CXXTokenTypeKeyword) &&
+								(pTokenBefore->pPrev->eKeyword == CXXKeywordDECLTYPE)
```

Comment:
Cannot we unify the two lines into one:
```
cxxTokenIsKeyword(pTokenBefore->pPrev, CXXKeywordDECLTYPE)
```
?

Reply:
Right!